aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -663,6 +698,78 @@ cunn_SoftMaxForward(outscalar_t *output, const scalar_t *input, int classes)
   }
 }
 
+template <int ILP, typename scalar_t, typename accscalar_t, typename outscalar_t, template <typename, typename, typename> class Epilogue>
+__global__ void
+cunn_SoftMaxForwardSmem(outscalar_t *output, const scalar_t *input, int classes)
```

Comment:
I'm not sure why we use `can_use_32bit_indexing` for some of the kernels, but not for others?
```suggestion
template <int ILP, typename scalar_t, typename accscalar_t, typename outscalar_t, template <typename, typename, typename> class Epilogue, typename index_t = int>
__global__ void
cunn_SoftMaxForwardSmem(outscalar_t *output, const scalar_t *input, index_t classes)
```